python/ray/serve/_private/common.py

```diff
@@ -141,6 +180,244 @@ def update(
             message=message,
         )
 
+    def update_message(self, message: str):
+        return self._update(message=message)
+
+    def transition(
+        self,
+        transition: DeploymentStatusTransition,
+        message: str = "",
+        is_within_autoscaling_bounds: bool = False,
+    ):
+        # If there was an unexpected internal error during reconciliation, set
+        # status to unhealthy immediately and return
+        if transition == DeploymentStatusTransition.INTERNAL_ERROR:
+            return self._update(
+                status=DeploymentStatus.UNHEALTHY,
+                status_trigger=DeploymentStatusTrigger.INTERNAL_ERROR,
+                message=message,
+            )
+
+        # If deployment is being deleted, set status immediately and return
+        if transition == DeploymentStatusTransition.DELETE:
```

Comment:
In cases when branches are mutex, let's specify those as `elif` to make it clear (reduces cognitive load of needing to keep multiple control-flows in memory when reading)